deepspeed/inference/v2/checkpoint/huggingface_engine.py

```diff
@@ -51,6 +53,7 @@ def _fetch_checkpoint_files(self):
                                                                "*.bin",
                                                                "*.json",
                                                                "*.pt",
+                                                               "*.safetensors",
```

Comment:
In the case that a model has both `*.bin` and `*.safetensors`, will this cause `snapshot_download` to download the model weights twice?

Reply:
Indeed there is this problem. Could we consider adding an extra parameter like hf transformers, safetensors=True/False, to adjust allow_pattern?

Reply:
Thank you for your feedback. I appreciate it. While developing the Safetensors model, my focus was primarily on downloading files with the `*.safetensors` extension. Upon further consideration, I have decided to introduce a new argument, `prioritize_safetensors: bool = True`, for this function. With this enhancement, the function will attempt to retrieve the prioritized format, but if unavailable, it will seamlessly fetch the alternative file type. This ensures flexibility, allowing users to choose their preferred model format.

Reply:
@jihnenglin thanks for updating this. I think we should try to make this a bit more automatic and seamless for the user. Rather than passing a `prioritize_safetensors` argument, we should automatically go the safetensor route if they are available and fallback to the .bin files if they are not. @jeffra thoughts?

Reply:
@mrwyattii Sounds great! I was not sure if you want to make it load `.safetensors` by default, so I kept both the options. But of course I'd be willing to make it more intuitive for the users. I will change it later.